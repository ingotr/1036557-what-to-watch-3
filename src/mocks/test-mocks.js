const FirstMovie = {
  TITLE: `Mock-test-The Grand Budapest Hotel`,
  GENRE: `Mock-test-Drama`,
  RELEASE_DATE: 2020,
};

const movies = [
  {
    id: `mock-test-011`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-012`,
    title: `Bohemian Rhapsody`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-013`,
    title: `Macbeth`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-014`,
    title: `Aviator`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-015`,
    title: `We need to talk about Kevin`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-016`,
    title: `What We Do in the Shadows`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-017`,
    title: `Revenant`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-018`,
    title: `Johnny English`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
];

const movieInfo = {
  title: `Mock-test-The Grand Budapest Hotel`,
  genre: `Mock-test-Drama`,
  year: 2017,
  poster: {
    big: `img/the-grand-budapest-hotel-poster.jpg`,
    bigAlt: `Mock-test-The Grand Budapest Hotel poster`,
  },
  rating: {
    score: 8.9,
    level: `Mocks-Very good`,
    count: 240,
  },
  director: `Mock-Wes Andreson`,
  description: [`Mock-test-In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave &#96s friend and protege.`,
    `Gustave prides himself on providing first-className service to the hotel &#96s guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave &#96s lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
  starring: [`Mock-test-Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
};

export {FirstMovie, movies, movieInfo};
