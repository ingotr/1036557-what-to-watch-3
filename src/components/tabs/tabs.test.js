import React from 'react';
import renderer from 'react-test-renderer';
import Tabs from './tabs.jsx';
import withActiveItem from '../../hocs/with-active-item/with-active-item.jsx';

import {film} from '../../mocks/test-mocks.js';

const TabsWrapped = withActiveItem(Tabs);

it(`VideoPlayer is rendered correctly`, () => {
  const tree = renderer
    .create(
        <TabsWrapped
          genre={film.genre}
          year={film.year}
          runtime={film.runtime}
          rating={film.rating}
          votes={film.votes}
          director={film.director}
          description={film.description}
          starring={film.starring}
          reviews={film.reviews}
        />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});

