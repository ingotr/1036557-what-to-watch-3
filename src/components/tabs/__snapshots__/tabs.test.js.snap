// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer is rendered correctly 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-rating movie-tab movie-tab--overview
        "
  >
    <div
      className="movie-rating__score"
    >
      8.9
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Mocks-Very good
      </span>
      <span
        className="movie-rating__count"
      >
        240
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text movie-tab movie-tab--overview
        "
  >
    <p>
      Mock-test-In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave &#96s friend and protege.
    </p>
    <p>
      Gustave prides himself on providing first-className service to the hotel &#96s guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave &#96s lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Mock-Wes Andreson
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Mock-test-Bill Murray, Edward Norton, Jude Law, Willem Dafoe
         and other
      </strong>
    </p>
  </div>,
  <div
    className="movie-card__text movie-card__row movie-tab movie-tab--details
        visually-hidden"
  >
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Director
        </strong>
        <span
          className="movie-card__details-value"
        >
          Mock-Wes Andreson
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Starring
        </strong>
        <span
          className="movie-card__details-value"
        >
          Mock-test-Bill Murray, 
Edward Norton, 
Jude Law, 
Willem Dafoe
        </span>
      </p>
    </div>
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Run Time
        </strong>
        <span
          className="movie-card__details-value"
        >
          1
          h 
          39
          m
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Genre
        </strong>
        <span
          className="movie-card__details-value"
        >
          Mock-test-Drama
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Released
        </strong>
        <span
          className="movie-card__details-value"
        >
          2017
        </span>
      </p>
    </div>
  </div>,
  <div
    className="movie-card__reviews movie-card__row movie-tab movie-tab--reviews
        visually-hidden"
  >
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Discerning travellers and Wes Anderson fans will luxuriate
       in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Kate Muir
            </cite>
            <time
              className="review__date"
              dateTime="2016-12-24"
            >
              December
               
              24
              , 
              2016
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8.9
        </div>
      </div>
      <br />
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Anderson's films are too precious for some, but for those
      of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Bill Goodykoontz
            </cite>
            <time
              className="review__date"
              dateTime="2015-11-18"
            >
              November
               
              18
              , 
              2015
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8
        </div>
      </div>
      <br />
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            I didn't find it amusing, and while I can appreciate the
      creativity, it's an hour and 40 minutes I wish I could take back.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Amanda Greever
            </cite>
            <time
              className="review__date"
              dateTime="2015-11-18"
            >
              November
               
              18
              , 
              2015
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8
        </div>
      </div>
      <br />
    </div>
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            The mannered, madcap proceedings are often delightful,
      occasionally silly, and here and there, gruesome and/or heartbreaking.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Matthew Lickona
            </cite>
            <time
              className="review__date"
              dateTime="2016-12-20"
            >
              December
               
              20
              , 
              2016
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          7.2
        </div>
      </div>
      <br />
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Paula Fleri-Soler
            </cite>
            <time
              className="review__date"
              dateTime="2016-12-20"
            >
              December
               
              20
              , 
              2016
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          7.6
        </div>
      </div>
      <br />
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Paula Fleri-Soler
            </cite>
            <time
              className="review__date"
              dateTime="2016-12-20"
            >
              December
               
              24
              , 
              2016
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          7
        </div>
      </div>
      <br />
    </div>
  </div>,
]
`;
